ast_tools/passes/util.py

```diff
@@ -82,14 +81,15 @@ def set_decorators(tree, decorators):
         return tree
 
 
-    @staticmethod
-    def get_name(node):
+    @classmethod
+    def lookup(cls, node, env):
         if isinstance(node, ast.Call):
-            assert isinstance(node.func, ast.Name)
-            return node.func.id
+            return cls.lookup(node.func, env)
+        elif isinstance(node, ast.Attribute):
+            return getattr(cls.lookup(node.value, env), node.attr)
```

Comment:
Does this need to be done recursively for `node.attr1.attr2`?

Reply:
node.value will be itself an attribute node, so the recursive call to `lookup` will handle it, unless you're asking if we should not handle the recursive attribute case?